ethereum/datastructures/src/main/java/tech/pegasys/teku/datastructures/state/AnchorPoint.java

```diff
@@ -30,33 +34,36 @@
  * Represents an "anchor" - a trusted, finalized (block, state, checkpoint) tuple from which we can
  * sync.
  */
-public class AnchorPoint {
+public class AnchorPoint extends StateAndBlockSummary {
   private final Checkpoint checkpoint;
-  private final SignedBeaconBlock block;
-  private final BeaconState state;
   private final boolean isGenesis;
-  private final SignedBlockAndState blockAndState;
 
   private AnchorPoint(
-      final Checkpoint checkpoint, final SignedBeaconBlock block, final BeaconState state) {
+      final Checkpoint checkpoint, final BeaconState state, final BeaconBlockSummary blockSummary) {
+    super(blockSummary, state);
     checkArgument(
-        block.getStateRoot().equals(state.hash_tree_root()), "Block and state must match");
-    checkArgument(checkpoint.getRoot().equals(block.getRoot()), "Checkpoint and block must match");
+        checkpoint.getRoot().equals(blockSummary.getRoot()), "Checkpoint and block must match");
 
     this.checkpoint = checkpoint;
-    this.block = block;
-    this.state = state;
     this.isGenesis = checkpoint.getEpoch().equals(UInt64.valueOf(Constants.GENESIS_EPOCH));
-    this.blockAndState = new SignedBlockAndState(block, state);
+  }
+
+  public static AnchorPoint create(
+      Checkpoint checkpoint, BeaconState state, Optional<SignedBeaconBlock> block) {
+    final BeaconBlockSummary blockSummary =
+        block
+            .map(BeaconBlockSummary.class::cast)
```

Comment:
This is potentially a bit heavy handed as it removes all compile time checking.  Can we get away with:
```suggestion
            .<BeaconBlockSummary>map(a -> a)
```

It doesn't read quite as well, but will give compile errors if `block` ever winds up not actually being a `BeaconBlockSummary`.

Reply:
good call 👍